### PR TITLE
Agressive Channeld Restart Testing

### PR DESCRIPTION
**Pull request by ddustin**
Repeatedly STFU and then TX_ABORT a channel to trigger the socket/fd flake issue.

Changelog-None

### CODE REVIEW

**Comment by rustyrussell** [comment]
This ran 50 times in a row for me without any error?

**Comment by ddustin** [comment]
It passed through CI no problem too -- it must be a Mac only problem

**Comment by cdecker** [comment]
Shall we merge this, or would you like to dig deeper and see what is going wrong with Mac? In the latter case I'd remove the milestone.

**Comment by ddustin** [comment]
Seems net positive to have it merged into the CI as a regression test

**Comment by ddustin** [comment]
Although this does bundle in the `stfu` and `abort` RPC commands which @niftynei had some design objections to in my Splice Script PR 🤔.

File: lightningd/channel_control.c [hunk 1]
```diff
@@ -2247,6 +2439,24 @@ static const struct json_command splice_signed_command = {
...
 AUTODATA(json_command, &splice_signed_command);
 
+static const struct json_command stfu_channels_command = {
+	"stfu_channels",
+	"channels",
+	json_stfu_channels,
+	"Put {channel_ids} channels into stfu mode and return their available"
+	" balance."
+};
+AUTODATA(json_command, &stfu_channels_command);
```
Comment (rustyrussell) [inline lightningd/channel_control.c:2449]: These are *definitely* dev commands!!

File: contrib/pyln-client/pyln/client/lightning.py [hunk 2]
```diff
@@ -1156,6 +1156,20 @@ def openchannel_abort(self, channel_id):
...
+        }
+        return self.call("stfu_channels", payload)
+
+    def abort_channels(self, channel_ids):
+        """ Abort multiple channels """
+        payload = {
+            "channel_ids": channel_ids,
+        }
+        return self.call("abort_channels", payload)
+
```
Comment (rustyrussell) [inline contrib/pyln-client/pyln/client/lightning.py:1172]: You don't need these wrappers, esp. for dev commands.  It automatically handles unknown RPCs (we mainly add them so you get nicer documentation and error detection).

File: tests/test_restart.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    l1.rpc.stfu_channels([chan_id])
+    l1.rpc.abort_channels([chan_id])
+    l1.daemon.wait_for_log(r'peer_in WIRE_CHANNEL_REESTABLISH')
+    l2.daemon.wait_for_log(r'peer_in WIRE_CHANNEL_REESTABLISH')
+
+    l1.rpc.stfu_channels([chan_id])
+    l1.rpc.abort_channels([chan_id])
+    l1.daemon.wait_for_log(r'peer_in WIRE_CHANNEL_REESTABLISH')
+    l2.daemon.wait_for_log(r'peer_in WIRE_CHANNEL_REESTABLISH')
```
Comment (rustyrussell) [inline tests/test_restart.py:25]: `for _ in range(20):` ?

**Review by rustyrussell — CHANGES_REQUESTED** [review]
Weird one.  You wrote this batch API and only ever use it for one channel.  Maybe just have a simpler, single-channel API?  Seems clearer, too...

File: lightningd/channel_control.c (hunk 1 above)
Comment (rustyrussell) [reply lightningd/channel_control.c:2449]: And since you only ever call it with one channel, `dev_quiesce` would work.

**Comment by ddustin** [comment]
> Weird one. You wrote this batch API and only ever use it for one channel. Maybe just have a simpler, single-channel API? Seems clearer, too...

The bulk commands were needed & implemented in #6980, this PR should remove its implementation and instead depend on #6980

File: tests/test_restart.py (hunk 3 above)
Comment (ddustin) [reply tests/test_restart.py:25]: Switched it out for a loop

**Comment by ddustin** [comment]
I removed the `stfu_channels` and `abort_channels` RPCs with the idea we can rebase off master once #6980 is merged.

I see two ways forward for these commands:
1) Expose low level splice commands to user: `stfu_channels`, `abort_channels`, `splice_init`, `splice_update`, `splice_signed` etc
2) Hide all these low level commands from user and expose a singular simple RPC like the `dev-splice` script command.

Doing something in between these two options doesn't make sense because a user doing a cross-splice of two channels needs a way to `stfu` both channels and potentially `abort` them. Prior to #6980 there was no way for a user to do so because `splice_init` entered `stfu` automatically.

File: lightningd/channel_control.c (hunk 1 above)
Comment (ddustin) [reply lightningd/channel_control.c:2449]: 🤔 Discussing the trade offs of exposing stfu / abort in this comment over here: https://github.com/ElementsProject/lightning/pull/7083#issuecomment-2467091184

**Comment by rustyrussell** [comment]
> I removed the `stfu_channels` and `abort_channels` RPCs with the idea we can rebase off master once #6980 is merged.
> 
> I see two ways forward for these commands:
> 
>     1. Expose low level splice commands to user: `stfu_channels`, `abort_channels`, `splice_init`, `splice_update`, `splice_signed` etc

I think this is the way.  Though make sftu/abort singular (you can simply call it multiple times).

**Comment by rustyrussell** [comment]
Trivial rebase...

**Review by rustyrussell — APPROVED** [review]

**Pull request merged**